test/notifications/display_test.dart

```diff
@@ -106,26 +108,54 @@ void main() {
   });
 
   group('NotificationDisplayManager show', () {
-    void checkNotification(MessageFcmMessage data, {
+    void checkNotification(List<MessageFcmMessage> messages, {
       required String expectedTitle,
       required String expectedTagComponent,
+      required bool expectedGroup,
     }) {
-      final expectedTag = '${data.realmUri}|${data.userId}|$expectedTagComponent';
-      final expectedGroupKey = '${data.realmUri}|${data.userId}';
-      final expectedId =
-        NotificationDisplayManager.notificationIdAsHashOf(expectedTag);
       const expectedIntentFlags =
         PendingIntentFlag.immutable | PendingIntentFlag.updateCurrent;
 
-      check(testBinding.androidNotificationHost.takeNotifyCalls())
-        ..length.equals(2)
-        ..containsInOrder(<Condition<AndroidNotificationHostApiNotifyCall>>[
-          (it) => it
+      final notifyCallsChecks = <Condition<AndroidNotificationHostApiNotifyCall>>[];
```

Comment:
nit:
```suggestion
      final notifyCallsChecks = <Condition<Object?>>[];
```

This doesn't directly affect any behavior, because the individual items already have type `Condition<Object?>` thanks to the annotation on the list literal that introduces them. But it's easier to think about when the different parts are consistent.